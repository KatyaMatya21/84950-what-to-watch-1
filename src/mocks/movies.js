export default [
  {
    'id': `1`,
    'title': `Fantastic Beasts: The Crimes of Grindelwald`,
    'src': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `2`,
    'title': `Bohemian Rhapsody`,
    'src': `img/bohemian-rhapsody.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `3`,
    'title': `Macbeth`,
    'src': `img/macbeth.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `4`,
    'title': `We need to talk about Kevin`,
    'src': `img/we-need-to-talk-about-kevin.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `5`,
    'title': `What We Do in the Shadows`,
    'src': `img/what-we-do-in-the-shadows.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `6`,
    'title': `Revenant`,
    'src': `img/revenant.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `7`,
    'title': `Johnny English`,
    'src': `img/johnny-english.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `8`,
    'title': `Shutter Island`,
    'src': `img/shutter-island.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `9`,
    'title': `Pulp Fiction`,
    'src': `img/pulp-fiction.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `10`,
    'title': `No Country for Old Men`,
    'src': `img/no-country-for-old-men.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `11`,
    'title': `Snatch`,
    'src': `img/snatch.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `12`,
    'title': `Moonrise Kingdom`,
    'src': `img/moonrise-kingdom.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `13`,
    'title': `Seven Years in Tibet`,
    'src': `img/seven-years-in-tibet.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `14`,
    'title': `Midnight Special`,
    'src': `img/midnight-special.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `15`,
    'title': `War of the Worlds`,
    'src': `img/war-of-the-worlds.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `16`,
    'title': `Dardjeeling Limited`,
    'src': `img/dardjeeling-limited.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `17`,
    'title': `Orlando`,
    'src': `img/orlando.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `18`,
    'title': `Mindhunter`,
    'src': `img/mindhunter.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `19`,
    'title': `Midnight Special`,
    'src': `img/midnight-special.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `20`,
    'title': `Aviator`,
    'src': `img/aviator.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];
