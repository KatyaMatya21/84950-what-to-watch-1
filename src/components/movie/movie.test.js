import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/movies';

import Movie from './movie.jsx';

const movie = movies[0];

describe(`Movie component`, () => {

  it(`Correctly renders`, () => {
    const MovieComponent = renderer
      .create(
          <Movie
            movie={movie}
            onMouseEnter={() => jest.fn()}
            onMouseOut={() => jest.fn()}
            isHovered={false}
          />,
          {
            createNodeMock: (element) => {
              if (element.type === `video`) {
                return {'current': {}};
              }
              return null;
            }
          }
      )
      .toJSON();
    expect(MovieComponent).toMatchSnapshot();
  });

});
