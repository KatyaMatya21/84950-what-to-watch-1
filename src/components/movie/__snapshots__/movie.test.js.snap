// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component Correctly renders 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      muted={true}
      poster="img/what-we-do-in-the-shadows.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      What We Do in the Shadows
    </a>
  </h3>
</article>
`;
